opensearch-operator/pkg/reconcilers/tls.go

```diff
@@ -259,6 +259,54 @@ func (r *TLSReconciler) adminSecretName() string {
 	return r.instance.Name + "-admin-cert"
 }
 
+// CRITEO WORKAROUND
+func (r *TLSReconciler) shouldCreateGlobalTransportCert(ca tls.Cert) (bool, error) {
```

Comment:
- if you pass secret name in argument, you can make this method agnostic to global or per node transport cert (potentially less rework for upstream contribution)
-  instead of returning bool, you could return loaded secret if cert exists and is valid (either via reference argument or return value). It will require less change on call site.

Reply:
> if you pass secret name in argument, you can make this method agnostic to global or per node transport cert (potentially less rework for upstream contribution)

Done and pushed; it is indeed clearer this way. I duplicated the function to make it easier to rebase/check for difference, but if we intend to make an upstream contribution it is indeed better this way

 > instead of returning bool, you could return loaded secret if cert exists and is valid (either via reference argument or return value). It will require less change on call site.

I'm not sure of the value it provides:

- for global transport cert, returning bool and having the function name start with "should" is easier to read, and matches upstream behavior. Changing it to return cert pointer or nil would only simplify a single line in a single codepath
- for per-node transport cert (which this PR doesn't touch), all node certs are packed into a single Kubernetes Secret and logic is quite different overall. This function (`shouldCreateClusterCert`) thus doesn't apply here:
    - I think this should be discussed with upstream maintainers, as we are not using this feature
    - one approach could be to only check bootstrap cert, since it is most of the time older than all the node certs
    - actually, _any_ cert properly signed by the CA cert seems to be usable as "transport" cert (OpenSearch transport port uses mutual certificate authentication). We now rely on this behavior in kubemanager to connect to the "transport" endpoint, using "admin" certificate as transport cert. Thus, approaches that renew transport certs more often than needed could be considered